### ISSUE TITLE
Selection background width based on window's width

### ISSUE THREAD

**Issue by bigrand** [philj56/tofi#75]
Using `selection-background-padding` allows you to apply padding to the selection background, but because each entry name may vary in characters length, it makes having a uniform selection background kind of difficult.

For example, If you try to make it uniform with percentages you may get in this situation:
![image](https://i.imgur.com/8B2ZMxj.png)

![image](https://user-images.githubusercontent.com/84900581/210854208-b34924c1-cd36-4a35-ae70-e73a6f6376a9.png)

As you can see, they aren't uniform at all, but the right side padding is 50%.

You can make it uniform, but that means using a really big padding so it gets cut off by the window padding, the downside is that it prevents you from using rounded borders, it either cuts off or you put a small enough padding that it's not longer uniform.

![image](https://i.imgur.com/Wd4vhUz.png)

**Comment by philj56** [comment]
This is standard behaviour for padding - it's extra size added to the existing size of the text, rather than the actual size of the background box.

I think the way to solve this would be to change to behaviour of passing `-1` to any of the `--*-background-padding` options. Currently that just makes the padding large enough such that it'll always go past the edge of the window, but it should be fairly straightforward to make it so that it just reaches the edge, allowing rounded corners to work. I'll look into it.

Just to make sure, when you say "right side padding is 50%", you mean `--padding-right=50%`? That'd work with this approach.

**Comment by bigrand** [comment]
Sorry for the delay in responding; I had some personal matters.

>  it should be fairly straightforward to make it so that it just reaches the edge, allowing rounded corners to work.

I believe it solves the "problem," but what if someone wants to make it uniform so that it takes up half the width of the window? For instance, if tofi adds support for maximum character length for entries, I could see someone wanting to make it uniform without taking up the entire width of the window.

Obviously, this isn't a big deal, and most people will want it to take up the entire width of the window.

> Are you referring to —padding-right=50%?

Yes.

**Comment by philj56** [comment]
Sorry for the huge delay, I started a new job and have been very busy! This should finally be done in 57d3a301b09c531972de4d012faafdc402f7b2d0 - specifying `-1` for padding just pads as far as possible (but no further) in that direction. For example,
```
tofi-run \
    --alternate-result-background 333 \
    --alternate-result-background-padding 5,-1,5,20 \
    --alternate-result-background-corner-radius 10 \
    --padding-right 50%
```
gives (slightly more than 50% apparent padding because this window is smaller than my screen):

![2023-02-26T11:58:06,188323764+00:00](https://user-images.githubusercontent.com/13610623/221409152-2a32def6-acb7-41d9-99fe-44eb27d50def.png)

**Comment by philj56** [comment]
This change has been released in [v0.9.0](https://github.com/philj56/tofi/releases/tag/v0.9.0).
